### ISSUE TITLE
Deployed page in production is broken

### ISSUE THREAD

**Issue by gmacario** [gmacario/gmacario.github.io#96]
As a consequence of merging #88 the contents on https://gmacario.github.io/ are not good

![image](https://user-images.githubusercontent.com/75182/118915546-7440c200-b92d-11eb-9634-3a2de31cfd25.png)

@GGn0 Do you have any suggestions?

**Comment by gmacario** [comment]
Trying to reproduce the bug locally as gmacario@gmpowerhorse (Ubuntu 20.04.2 LTS)

```bash
git clean -fdx
git reset --hard
git status
git describe --tags
```

Result

```bash
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)$ git clean -fdx
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)$ git reset --hard
HEAD is now at b7dafdb Merge pull request #88 from gmacario/migrate_to_nextJS
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)$ git status
On branch fix/issue-96
nothing to commit, working tree clean
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)$ git describe --tags
broken-issue-96
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)$
```

Building static pages

```bash
npm install
npm test
```

Result

```bash
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)$ npm install
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
npm WARN my-blog@1.0.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 413 packages from 338 contributors and audited 415 packages in 11.239s

63 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)*$ npm test
TODO
```

**Comment by gmacario** [comment]
Let's create a test branch with the updated `package-lock.json`

```bash
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)*$ npm --version
6.14.13
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)*$ git commit . -sm "package-lock.json generated by npm 6.14.13"
[fix/issue-96 0296876] package-lock.json generated by npm 6.14.13
 1 file changed, 3209 insertions(+), 8456 deletions(-)
 rewrite package-lock.json (60%)
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)$ git push -u origin fix/issue-96 
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 8 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (3/3), 730 bytes | 730.00 KiB/s, done.
Total 3 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
remote: 
remote: Create a pull request for 'fix/issue-96' on GitHub by visiting:
remote:      https://github.com/gmacario/gmacario.github.io/pull/new/fix/issue-96
remote: 
To https://github.com/gmacario/gmacario.github.io.git
 * [new branch]      fix/issue-96 -> fix/issue-96
Branch 'fix/issue-96' set up to track remote branch 'fix/issue-96' from 'origin'.
gmacario@gmpowerhorse:~/github/gmacario/gmacario.github.io (fix/issue-96)$
```

**Comment by gmacario** [comment]
The issue was not fixed, reopening.

CC: @gteti

**Comment by gteti** [comment]
Missing some features. Running 
> npm test
Returns 
> ERR! missing script test

**Comment by gmacario** [comment]
Let's make a game: the first one who will submit a viable PR to fix #96 will get a free :coffee:

**Comment by gmacario** [comment]
@GGn0 I checked out branch gh-pages and the generated static site is broken as well:

![image](https://user-images.githubusercontent.com/75182/118961153-4c208580-b964-11eb-8074-bc894b642689.png)

**Comment by GGn0** [comment]
I just noticed that the site only works in develop mode
```sh
npm run dev
```
I'm trying to figure out why

regarding your first picture
![image](https://user-images.githubusercontent.com/44038661/119125330-f1555f80-ba31-11eb-9d4e-25cd82ec1015.png)
Have you set the source of github pages to the gh-pages branch?

After that you should at least see the broken site

**Comment by gmacario** [comment]
@GGn0 after merging #98 here is what I get on https://gmacario.github.io/

![image](https://user-images.githubusercontent.com/75182/119228476-75cbdf00-bb13-11eb-996f-1bc8a81fbcda.png)

This may be the consequence of today 👩‍⚕️

**Comment by gmacario** [comment]
Concerning your previous question

> Have you set the source of github pages to the gh-pages branch?

Here are current https://github.com/gmacario/gmacario.github.io/settings/pages

![image](https://user-images.githubusercontent.com/75182/119228573-e70b9200-bb13-11eb-863d-052e066a4770.png)

Let's switch to Branch: `gh-pages` instead

**Comment by gmacario** [comment]
I am afraid that #96 is harder to fix than expected 😢 

Result of the last deployment to https://gmacario.github.io/ after merging #99

![Schermata 2021-05-22 alle 15 59 21](https://user-images.githubusercontent.com/75182/119229170-eb857a00-bb16-11eb-9482-f2b57238928d.png)

**Comment by gmacario** [comment]
@GGn0 I suspect that CSS and/or other static assets are not copied to the gh-pages branch

PTAL https://github.com/gmacario/gmacario.github.io/tree/gh-pages

**Comment by GGn0** [comment]
try merging #100 
it is probably due to the .nojekyll file being in the wrong directory

i've changed the workflow file

check [ggn0.github.io](https://ggn0.github.io) where i temporarily pushed the updated code

**Comment by gmacario** [comment]
It works!

https://github.com/gmacario/gmacario.github.io/issues/96#issuecomment-844768865

> Let's make a game: the first one who submits a viable PR to fix #96 will get a free ☕

@GGn0 You submitted two PR (#98 and #100), but technically you fixed only #96, so you have won only one ☕  according to the rules stated in https://github.com/gmacario/gmacario.github.io/issues/96#issuecomment-844768865 🤣
